### PR TITLE
Fix CS to PSR-12 standard

### PR DESCRIPTION
**Pull request by mvorisek**
the only changed files are:
- `.gitignore`
- `.php_cs.dist`
- `composer.json`

the php files are unmodified and strictly only fixed by CS Fixer

to prevent conflicts, this PR should be merged immediatelly

PS: feel free to rerun CS Fixer, but must be run twice to stabilize and each run takes almost a day (https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/5279)

### CODE REVIEW

File: .php_cs.dist [hunk 1]
```diff
@@ -0,0 +1,75 @@
+<?php
+
+$finder = PhpCsFixer\Finder::create()
```
Comment (keradus) [inline .php_cs.dist:3]: pssst, you may disable autogenerated, 1MB+ files that are too slow to be processed

File: .php_cs.dist [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        'yoda_style' => [
+            'equal' => false,
+            'identical' => false,
+        ],
+        'native_function_invocation' => false,
+        'void_return' => false,
+        'blank_line_before_statement' => [
+            'statements' => ['break', 'continue', 'declare', 'return', 'throw', 'exit'],
+        ],
+        'combine_consecutive_issets' => false,
```
Comment (keradus) [inline .php_cs.dist:35]: if there is a need to disable 30 rules, maybe it's better to explicitly enable full rulesets like PSR2 and maybe few rules on top of it?

Comment (mvorisek) [reply .php_cs.dist:35]: @keradus I have simple config - https://github.com/mvorisek/TCPDF/tree/psr12_cs_extended - as far I as remember there is one rule that does not finish at all. I belive, in CS fixer, there are several small issues with extreme/exponencial complexity. That is why I opened https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/5279

Comment (mvorisek) [reply .php_cs.dist:35]: @keradus the issue is with `array_indentation`, when not disabled, the CS Fixer never finish (I waited more than 24 hours)

Comment (keradus) [reply .php_cs.dist:35]: https://github.com/tecnickcom/TCPDF/pull/300#discussion_r532117195

Comment (mvorisek) [reply .php_cs.dist:35]: I do not argue the files are large. But `array_indentation` takes at least 100x more time than the other rules. So I think this is an issue.

Comment (keradus) [reply .php_cs.dist:35]: as i shared in Fixer repo - if you want, do not hesitate to send a PR to make it faster. if you don't want, don't expect someone will do it for you. same time, you got provided an alternative approach - don't fix autogenerated files that you also do not review.

**Review by nicolaasuni — COMMENTED** [review]
It looks like that this PR includes breaking changes for older php versions.
I'd rather keep this version as it is and focus on the new tc-lib-pdf library.

**Comment by mvorisek** [comment]
> 
> 
> It looks like that this PR includes breaking changes for older php versions.
> I'd rather keep this version as it is and focus on the new tc-lib-pdf library.

That is correct. TCPDF is hovewer used in a lot of projects and this PR improves maintainability of this project.

I belive dropping support for versions lower than PHP 7.1 is ok for two reasons:
- lower versions of TCPDF can still of course be installed
- PHP 7.1 itself is deprecated and does not receive any updates

**Comment by mvorisek** [comment]
@williamdes what about landing this into WarnockPDF? Maybe even rebase the commits so this is the first commit so the new/fork changes made are easily diffable.

**Comment by williamdes** [comment]
> @williamdes what about landing this into WarnockPDF? Maybe even rebase the commits so this is the first commit so the new/fork changes made are easily diffable.

Good idea, if you can apply https://github.com/wdes/coding-standard that would be awesome !

> Maybe even rebase the commits so this is the first commit so the new/fork changes made are easily diffable.

You can rebase, the history is the same but I did some cleaning recently

**Comment by mvorisek** [comment]
Closing here, if this project is gonna be supported someday, backport the changes from https://github.com/code-lts/WarnockPDF fork

**Pull request closed without merging**